src/core/ledger/grainIntegration.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import {Ledger, type PayoutAddress, type CurrencyId} from "./ledger";
+import {type Distribution} from "./distribution";
+import {getDistributionBalances} from "./distributionSummary/distributionSummary.js";
+import {type Currency} from "./currency.js";
+import {type IdentityId} from "../identity";
+import * as G from "./grain";
+
+type Transfer = {|
+  amount: G.Grain,
+  memo: string,
+|};
+
+export type AccountDistributions = Map<PayoutAddress, G.Grain>;
+export type PayoutAddressToId = Map<PayoutAddress, IdentityId>;
+export type TransferredGrain = Map<PayoutAddress, Transfer>;
```

Comment:
nit: Is a map really what you want here? I don't see individual look ups happening, and this seems like something that would want to be serialization-friendly and array.map-friendly (optimize for transformability over lookup).

Reply:
Maps are versatile, and simple enough to iterate over. If an array is needed, using `Array.from()` will suffice on the grain integration side. 

This data structure will strictly be accessed by the Grain Integration in memory,, so a Map is okay. If we start passing it through an RPC or some other transport that requires serilaization for some reason, we'll have to revisit it, course.